Automattic/jetpack
# Blocks: update all blocks to use namespaces and constants for consistency

Fixes #14843

#### Changes proposed in this Pull Request:

This PR does not introduce any new functionality or changes anything in the way the blocks work today. It does, however, change the way blocks are registered to bring all blocks to the same level and be consistent across all blocks. It should help new Jetpack contributors who work on blocks, by giving them one approach they can follow to register a new block.

- I've updated the template used when using WP Cli to create a new block. This is work that was already started in #14728.
- I've updated all blocks to be under the `Automattic\Jetpack` namespace (vs. just `Jetpack` before, or no namespace at all). Adding the vendor name to the namespace is how we work with all other namespaces in Jetpack. **Part of me wonders if we shouldn't introduce one more level there, so all blocks live under an `Automattic\Jetpack\Blocks` or `Automattic\Jetpack\Extensions` namespace. It may make things easier if one day we move those blocks into a package.**


#### Testing instructions:

* You should be able to use all those blocks with no issues or notices in your logs:
    - Amazon (Beta)
    - Business Hours
    - Calendly
    - Eventbrite
    - Gif
    - Google Calendar
    - Instagram Gallery (Beta)
    - Mailchimp
    - Map
    - Markdown
    - Opentable
    - Pinterest
    - Podcast Player (Beta)
    - Rating Star
    - Repeat Visitor
    - Revue
    - Slideshow
    - Tiled Gallery
    - WordAds

#### Proposed changelog entry for your changes:

* N/A
